src/validator.ts

```diff
@@ -26,24 +26,30 @@ export class Validator {
         const validate = ajv.compile(schema);
         const valid = validate(gitLabCiConfig);
         if (!valid) {
-            writeStreams.stderr(chalk`Invalid .gitlab-ci.yml configuration!\n`);
-
             const betterErrors = betterAjvErrors({
                 data: gitLabCiConfig,
                 errors: validate.errors,
             });
 
-            const errors = betterErrors.map((e => (chalk`\t• {redBright ${e.message}} at {blueBright ${e.path}}`)));
-            writeStreams.stderr(errors.splice(0, MAX_ERRORS).join("\n"));
-            if (errors.length > MAX_ERRORS) {
-                writeStreams.stderr(`\t... and ${errors.length - MAX_ERRORS} more`);
-            }
+            let index = 0;
+            const errMessages: string[] = [];
+            betterErrors.some(e => {
+                errMessages.push(chalk`\t• {redBright ${e.message}} at {blueBright ${e.path}}`);
+                index++;
+                if (index == MAX_ERRORS) {
+                    errMessages.push(chalk`\t... and ${betterErrors.length - MAX_ERRORS} more`);
+                }
+                return index > MAX_ERRORS - 1 ;
+            });
+
+            assert(valid, chalk`
+{reset Invalid .gitlab-ci.yml configuration!
+${errMessages.join("\n")}
 
-            writeStreams.stderr(chalk`\n\nFor further troubleshooting, consider either of the following:
+For further troubleshooting, consider either of the following:
 \t• Copy the content of {blueBright ${terminalLink(".gitlab-ci-local/expanded-gitlab-ci.yml", pathToExpandedGitLabCi)}} to the ${terminalLink("pipeline editor", "https://docs.gitlab.com/ee/ci/pipeline_editor/")} to debug it
-\t• Use --json-schema-validation=false to disable schema validation (not recommended)
+\t• Use --json-schema-validation=false to disable schema validation (not recommended)}
 `);
-            process.exit(1);
```

Comment:
refactored to use asserts to be more consistent 

otherwise when unit test fails from json validation check, it's less clear because the failed unit test